src/engine/IndexScan.h

```diff
@@ -93,6 +93,10 @@ class IndexScan : public Operation {
 
   bool knownEmptyResult() override { return getExactSize() == 0; }
 
+  bool isIndexScanWithNumVariables(size_t numVariables) const override {
+    return getResultWidth() == numVariables;
```

Comment:
As stated, this has to use the `numVariables()` function, not `getResultWidth()` because the latter might be wrong because of additional columns.